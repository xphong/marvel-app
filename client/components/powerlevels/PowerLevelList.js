import PropTypes from 'prop-types';
import React, { Component } from 'react';

import PowerLevelRow from './PowerLevelRow';

export default class PowerLevelList extends Component {
  constructor(props) {
    super(props);
  }

  renderPowerLevelRow(characterData, index) {
    return <PowerLevelRow key={characterData.Name} characterData={characterData} index={index} />;
  }

  render() {
    const { data } = this.props.powerLevels;

    return (
      <table className="ui table">
        <thead>
          <tr>
            <th>Character Name</th>
            <th>Average Power Level</th>
            <th>More Info Link</th>
          </tr>
        </thead>
        <tbody>
          {data.map(this.renderPowerLevelRow)}
        </tbody>
      </table>
    );
  }
}

PowerLevelList.propTypes = {
  powerLevels: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired
};
