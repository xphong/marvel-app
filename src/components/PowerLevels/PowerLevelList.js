import React, { Component, PropTypes } from 'react';

import PowerLevelRow from './PowerLevelRow';

export default class PowerLevelList extends Component {
  constructor(props){
    super(props);
  }

  renderPowerLevelRow(characterData) {
    return <PowerLevelRow key={characterData.Name} characterData={characterData} />;
  }

  render() {
    const {data} = this.props.powerLevels;

    return (
      <table className='ui table'>
        <thead>
          <tr>
            <th>Character Name</th>
            <th>Average Power Level</th>
            <th>More Info Link</th>
          </tr>
        </thead>
        <tbody>
          {data.map(this.renderPowerLevelRow)}
        </tbody>
      </table>
    );
  }
}

PowerLevelList.propTypes = {
  powerLevels: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired
};
