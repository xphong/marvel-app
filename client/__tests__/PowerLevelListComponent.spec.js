import React from 'react';
import ReactDOM from 'react-dom';

import PowerLevelList from '../components/powerlevels/PowerLevelList';

describe('PowerLevelList Component', () => {
  it('should render PowerLevelList', () => {
    const props = {
      actions: {},
      powerLevels: {
        data: [{
          Name: 'Storm',
          AveragePowerLevel: 9000
        }]
      }
    };

    const wrapper = shallow(<PowerLevelList {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
