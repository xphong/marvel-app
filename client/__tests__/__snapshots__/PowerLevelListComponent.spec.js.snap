// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PowerLevelList Component should render PowerLevelList 1`] = `
<table
  className="ui table"
>
  <thead>
    <tr>
      <th>
        Character Name
      </th>
      <th>
        Average Power Level
      </th>
      <th>
        More Info Link
      </th>
    </tr>
  </thead>
  <tbody>
    <PowerLevelRow
      characterData={
        Object {
          "AveragePowerLevel": 9000,
          "Name": "Storm",
        }
      }
      index={0}
      key="Storm"
    />
  </tbody>
</table>
`;
